modules/_labs/select/react/lib/SelectBase.tsx

```diff
@@ -363,105 +326,140 @@ export default class SelectBase extends React.Component<SelectBaseProps> {
+        <SelectMenu
+          aria-activedescendant={options[focusedOptionIndex].id}
+          aria-labelledby={ariaLabelledBy}
+          aria-required={ariaRequired || required}
+          buttonRef={buttonRef}
+          id={menuId}
           error={error}
-          grow={grow}
-          isMenuHidden={isMenuHidden}
-          isMenuHiding={isMenuHiding}
+          menuRef={menuRef}
+          onBlur={onMenuBlur}
           onKeyDown={onKeyDown}
-          // Prevent Firefox from triggering click handler on spacebar during
-          // type-ahead when the menu is closed (and, thus, incorrectly displaying
-          // the menu)
-          onKeyUp={e => {
-            e.preventDefault();
-          }}
-          ref={buttonRef}
-          type="button"
-          value={selectedOptionValue}
-          {...elemProps}
+          onCloseOnEscape={onMenuCloseOnEscape}
+          placement={menuPlacement}
+          shouldAutoFlip={shouldMenuAutoFlip}
+          shouldAutoFocus={shouldMenuAutoFocus}
+          visibility={menuVisibility}
         >
-          {selectedOptionLabel}
-        </SelectButton>
-        <SelectInput onChange={onChange} ref={inputRef} type="text" value={selectedOptionValue} />
-        {!isEmpty && !isMenuHidden && (
-          <SelectMenu
-            aria-activedescendant={options[focusedOptionIndex].id}
-            aria-labelledby={ariaLabelledBy}
-            aria-required={ariaRequired || required}
-            buttonRef={buttonRef}
-            id={this.menuId}
-            error={error}
-            isFlipped={isMenuFlipped}
-            isHidden={isMenuHidden}
-            isHiding={isMenuHiding}
-            menuRef={this.menuRef}
-            onBlur={onMenuBlur}
-            onKeyDown={onKeyDown}
-            onCloseOnEscape={onMenuCloseOnEscape}
-            shouldAnimate={shouldMenuAnimate}
-            shouldAutoFlip={shouldMenuAutoFlip}
-            shouldAutoFocus={shouldMenuAutoFocus}
-          >
-            {this.renderOptions(renderOption)}
-          </SelectMenu>
-        )}
-        <SelectMenuIcon
-          className="menu-icon"
-          icon={caretDownSmallIcon}
-          color={disabled ? colors.licorice100 : colors.licorice200}
-          colorHover={disabled ? colors.licorice100 : colors.licorice500}
-          size={menuIconSize}
-        />
-      </SelectWrapper>
-    );
-  }
-}
+          {renderOptions(renderOptionFunction)}
+        </SelectMenu>
+      )}
+      <SelectMenuIcon
+        className="menu-icon"
+        icon={caretDownSmallIcon}
+        color={disabled ? colors.licorice100 : colors.licorice200}
+        colorHover={disabled ? colors.licorice100 : colors.licorice500}
+        size={menuIconSize}
+      />
+    </SelectWrapper>
+  );
+};
+
+SelectBase.defaultProps = {
+  focusedOptionIndex: 0,
+  isMenuFlipped: false,
+  menuPlacement: 'bottom',
+  menuVisibility: 'closed',
+  shouldMenuAutoFlip: true,
+  shouldMenuAutoFocus: true,
+};
```

Comment:
Default parameters sounds great. Looks like we already set the precedent in #525 (Labs Select wasn't included in the refactor since it was still in progress).

Besides, it looks like [`defaultProps` will be deprecated](https://twitter.com/dan_abramov/status/1133878326358171650) at some point.